frontend/app/src/components/history/type.d.ts

```diff
@@ -1,9 +1,11 @@
+import {VueConstructor} from "vue";
 import { TradeLocation } from '@/services/history/types';
 
 export type TradeLocationData = {
   readonly identifier: TradeLocation;
   readonly name: string;
   readonly icon: string;
   readonly imageIcon: boolean;
+  readonly component: VueConstructor | boolean;
```

Comment:
Out of curiosity Is there any specific reason why the `| boolean` approach was selected instead of making it optional and leaving the value undefined if there is no component?

Reply:
@kelsos I wasn't sure of the proper syntax.  I will change it to read `readonly component?: VueConstructor;`